File: 02-FEA-1D/Matlab-Code/MainFile.m

clc; clear all;

%% ME 471 / AE 420 / 
% Programming Assignment #2: 1D elastostatic FE code
% Prof. Mariana Silva

% Variables definition

% nnode_ele, the (integer) number of nodes per element
% node_dof, the (integer) number of degrees of freedom per node
% edof, the (integer) number of degrees of freedom per element

NNODE_ELE = 2; %Number of nodes per element
DOF_NODE = 1; %Number of degrees of freedom per node
EDOF = NNODE_ELE*DOF_NODE; %number of degrees of freedom per element

filename = 'input.dat'; 

%read
[COORDS,N_NODE,N_ELEM,N_LOAD,N_PRE_DISP,...
    ELEM_NODE,ELEM_STIFF,...
    ELEM_LOAD,LOAD_NODE,LOAD_VAL,DISP_NODE,DISP_VAL] = ReadInput(filename,NNODE_ELE);



%initialize
[EQ_NUM,N_DOF] = Initialize(N_NODE,N_PRE_DISP,DISP_NODE);

%ASSEMBLE
[KPP,KPF,KFF,KFP,PP,PF,UP] = Assemble(COORDS,N_ELEM,N_LOAD,N_PRE_DISP,ELEM_NODE,ELEM_STIFF,ELEM_LOAD,LOAD_NODE,LOAD_VAL,DISP_NODE,DISP_VAL,EQ_NUM,N_DOF);


%SOOOOOLVE!
[UUR,PUR,UF,PP] = Solve (N_NODE,KPP,KPF,KFF,KFP,PP,PF,UP,EQ_NUM);

%priiiiint~

PrintOutput(EQ_NUM,KPP,KPF,KFP,KFF,UF,PP,UUR,PUR)

File: 03-FEA-1D-IsoparametricElements/C-Code/MainFile.cpp
using namespace std;
#include <MainFile.h>

// ME 471 / AE 420
// Programming Assignment #3: 1D elastostatic FE code 
// includes: 
// - isoparametric linear and quadratic elements
// - numerical integration using Ng = 1,2,3,4 gauss points
// - DLOAD function for user-defined body load b(x)
// - UMAT function for user-defined elasticity modulus E(x)

/* Write your main code here */

int main()
{

  int nnode_ele; //nnode_ele: number of nodes per element
  int dof_node;  //dof_node: number of degree of freedom per node
  int edof;      //edof: number of degree of freedom per element
  int n_dof;
  int total_dof;

  int n_node;
  int n_elem;
  int n_load;
  int n_pre_disp;

  arma::imat elem_node;
  arma::colvec elem_stiff;
  arma::icolvec force_node;
  arma::colvec force_val;
  arma::icolvec disp_node;
  arma::colvec disp_val;

  arma::imat eq_num;
  arma::colvec Pf;
  arma::colvec Pp;
  arma::colvec Uf;
  arma::colvec Up;
  arma::colvec Uur;
  arma::colvec Pur;
  arma::mat Kff;
  arma::mat Kfp;
  arma::mat Kpf;
  arma::mat Kpp;

  // =========================================================
  // DO NOT MODIFY THE LINE BELOW!!
  //Autograding script will search for this variable definition
  string filename = "input.dat";
  // =========================================================
 
  /*=========================================================;
  %                 Complete your FEA code                   ;
  %=========================================================*/
  
 /*=========================================================;
  %                 Write Output Files                      ;
  %=========================================================*/ 
  PrintOutput( n_dof, total_dof,  eq_num, Kff,  Kpp,  Kpf,  Kfp, Uf , Pp, Uur, Pur);

  return(0);
}



